### DISCUSSION TITLE
v9-RC: Content published notification on replica servers

### DISCUSSION THREAD

**Discussion by vsilvar** [umbraco/Umbraco-CMS#10649] [Features and ideas]
By design, I think, `ContentPublishedNotification` is only triggered on the master server.
This means that if, for example, a notification is needed to clear some sort of custom view cache, we need to rely on both `ContentCacheRefresherNotification` and `MediaCacheRefresherNotification`.

While this works, it's not ideal, as it seems that `ContentCacheRefresherNotification` not only triggers for published content but for simple saves as well, with no real way to discern them.

Is there any way to achieve this?
Or even better, could a notification be triggered when the Umbraco `CachedPartial` gets cleared, so we could hook into a single point?

**Comment by nul800sebastiaan** [comment]
Hi @vsilvar - I'm not sure what you mean here? `CachedPartial` is not an Umbraco thing but an MVC thing, don't think we can clear that?

**Reply by nul800sebastiaan** [reply]
Ah no, it looks like we made our own `CachedPartial` extensions as well, which is what I'm assuming is what you're using.

If you have some example code it might be easier to understand what you're trying to do and what isn't working for you.

**Reply by vsilvar** [reply]
Hi @nul800sebastiaan,
I want to clear my own custom cache when content is published in Umbraco.

Currently, if I want that to work in any of the replica servers, I need to subscribe to the `ContentCacheRefresherNotification`.
The problem with this is that it also triggers on content saves without publishing, ideally I wouldn't clear the cache in those situations, but I have no way of differentiating a save from a publish.

You guys already clear your own `CachedPartial` in all the right situations, so a possible solution would be for Umbraco to trigger a new notification `CachedPartialCacheCleared` that I could then subscribe to.

But this is just a suggestion, in the end, the problem is that currently, there is no way for me to clear my custom cache only on content publish.

**Reply by bergmania** [reply]
IMO it could make sense to publish a notification as you say, I'm, just unsure where it makes the most sense.. The `ClearPartialViewCache` is just an extension method, which is not suited for handling this too. 

I guess this should be moved to a service and handle both the clear and Get/Add. Basically used everywhere the `CoreCacheHelperExtensions.PartialViewCacheKey` is used, and change this const to be private.
